src/anomalib/engine/engine.py

```diff
@@ -324,8 +324,12 @@ def _setup_trainer(self, model: AnomalyModule) -> None:
         # Setup anomalib callbacks to be used with the trainer
         self._setup_anomalib_callbacks()
 
-        # Temporarily set devices to 1 to avoid issues with multiple processes
-        self._cache.args["devices"] = 1
+        # TODO(ashwinvaidya17, djdameln, samet-akcay): Add Multi-GPU support to Anomalib
+        # https://github.com/openvinotoolkit/anomalib/issues/1449
+        devices = self._cache.args.get("devices")
+        if devices and str(devices).count(","):
```

Comment:
What happens if someone passes devices `=2`? If I am not wrong, it will still start distributed training.

Reply:
Good catch! I've added few more logic to cover some other edge cases.